rules/CodingStyle/Rector/ClassMethod/MakeInheritedMethodVisibilitySameAsParentRector.php

```diff
@@ -79,51 +78,56 @@ protected function run()
      */
     public function getNodeTypes(): array
     {
-        return [ClassMethod::class];
+        return [Class_::class];
     }
 
     /**
-     * @param ClassMethod $node
+     * @param Class_ $node
      */
-    public function refactor(Node $node): ?Node
+    public function refactorWithScope(Node $node, Scope $scope): ?Node
     {
-        if ($node->isMagic()) {
-            return null;
-        }
-
-        $scope = $node->getAttribute(AttributeKey::SCOPE);
-        if (! $scope instanceof Scope) {
-            // possibly trait
-            return null;
-        }
+        $hasChanged = false;
 
         $classReflection = $scope->getClassReflection();
         if (! $classReflection instanceof ClassReflection) {
             return null;
         }
 
-        /** @var string $methodName */
-        $methodName = $this->getName($node->name);
-
-        foreach ($classReflection->getParents() as $parentClassReflection) {
-            $nativeClassReflection = $parentClassReflection->getNativeReflection();
+        if ($classReflection->getParents() === []) {
+            return null;
+        }
 
-            // the class reflection aboves takes also @method annotations into an account
-            if (! $nativeClassReflection->hasMethod($methodName)) {
+        foreach ($node->getMethods() as $classMethod) {
+            if ($classMethod->isMagic()) {
                 continue;
             }
 
-            $parentReflectionMethod = $nativeClassReflection->getMethod($methodName);
-            if ($this->isClassMethodCompatibleWithParentReflectionMethod($node, $parentReflectionMethod)) {
-                return null;
-            }
+            /** @var string $methodName */
+            $methodName = $this->getName($classMethod->name);
 
-            if ($this->isConstructorWithStaticFactory($node, $methodName)) {
-                return null;
-            }
+            foreach ($classReflection->getParents() as $parentClassReflection) {
+                $nativeClassReflection = $parentClassReflection->getNativeReflection();
```

Comment:
The `$classReflection->getParents()` need to be saved to variable before loop to increase performance